src/planwise/component/scenarios.clj

```diff
@@ -36,13 +36,42 @@
   (let [changeset (filter #(-> % :initial nil?) changeset)]
     (apply +' (map :investment changeset))))
 
+(defn- count-actions
+  [changeset]
+  (reduce
+   (fn [counters change]
+     (letfn [(action-to-index [action]
+               (case action
+                 "create-provider" 0
+                 "upgrade-provider" 1
+                 "increase-provider" 2))]
+       (update-in counters [(action-to-index (:action change))] inc)))
+   [0 0 0]
+   changeset))
+
+(def action-labels ["create" "upgrade" "increase capacity for"])
```

Comment:
If using `frequencies`, you'd need to change this to a map from `:action` into label. You will lose the ordering, but you can make it explicit in `build-summary` later.